x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/PreviewTransformAction.java

```diff
@@ -92,20 +111,36 @@ public static Request fromXContent(final XContentParser parser) throws IOExcepti
 
         @Override
         public ActionRequestValidationException validate() {
+            // TODO: Make sure ids do not clash
+
+
```

Comment:
I am not 100% sure that is a big issue. I don't think the `_transforms/<id>/_preview` should allow a body at all. Consequently, there shouldn't be any IDs clashing.